FirebasePerformance/Sources/FPRNanoPbUtils.m

```diff
@@ -0,0 +1,452 @@
+  //  NSMutableArray<FPRMSGIosMemoryReading *> *memoryReadings =
+  //      [[NSMutableArray<FPRMSGIosMemoryReading *> alloc] init];
+  //  [gaugeData enumerateObjectsUsingBlock:^(id obj, NSUInteger idx, BOOL *stop) {
+  //    if ([obj isKindOfClass:[FPRCPUGaugeData class]]) {
+  //      FPRCPUGaugeData *gaugeData = (FPRCPUGaugeData *)obj;
+  //      FPRMSGCpuMetricReading *cpuReading = [FPRMSGCpuMetricReading message];
+  //      cpuReading.clientTimeUs = gaugeData.collectionTime.timeIntervalSince1970 * USEC_PER_SEC;
+  //      cpuReading.systemTimeUs = gaugeData.systemTime;
+  //      cpuReading.userTimeUs = gaugeData.userTime;
+  //      [cpuReadings addObject:cpuReading];
+  //    }
+  //
+  //    if ([obj isKindOfClass:[FPRMemoryGaugeData class]]) {
+  //      FPRMemoryGaugeData *gaugeData = (FPRMemoryGaugeData *)obj;
+  //      FPRMSGIosMemoryReading *memoryReading = [FPRMSGIosMemoryReading message];
+  //      memoryReading.clientTimeUs = gaugeData.collectionTime.timeIntervalSince1970 *
+  //      USEC_PER_SEC; memoryReading.usedAppHeapMemoryKb =
+  //      (int32_t)BYTES_TO_KB(gaugeData.heapUsed); memoryReading.freeAppHeapMemoryKb =
+  //      (int32_t)BYTES_TO_KB(gaugeData.heapAvailable); [memoryReadings addObject:memoryReading];
+  //    }
+  //  }];
+  //  gaugeMetric.cpuMetricReadingsArray = cpuReadings;
+  //  gaugeMetric.iosMemoryReadingsArray = memoryReadings;
+  //  return gaugeMetric;
+  firebase_perf_v1_GaugeMetric gaugeMetric = firebase_perf_v1_GaugeMetric_init_default;
+  return gaugeMetric;
+}
+
+firebase_perf_v1_ApplicationProcessState ApplicationProcessState(FPRTraceState state) {
+  //  FPRMSGApplicationProcessState processState =
+  //      FPRMSGApplicationProcessState_ApplicationProcessStateUnknown;
+  //  switch (state) {
+  //    case FPRTraceStateForegroundOnly:
+  //      processState = FPRMSGApplicationProcessState_Foreground;
+  //      break;
+  //
+  //    case FPRTraceStateBackgroundOnly:
+  //      processState = FPRMSGApplicationProcessState_Background;
+  //      break;
+  //
+  //    case FPRTraceStateBackgroundAndForeground:
+  //      processState = FPRMSGApplicationProcessState_ForegroundBackground;
+  //      break;
+  //
+  //    default:
+  //      break;
+  //  }
+  firebase_perf_v1_ApplicationProcessState processState =
+      firebase_perf_v1_ApplicationProcessState_BACKGROUND;
+  return processState;
+}
+
+#ifdef TARGET_HAS_MOBILE_CONNECTIVITY
+CTTelephonyNetworkInfo *NetworkInfo() {
+  static CTTelephonyNetworkInfo *networkInfo;
+  static dispatch_once_t onceToken;
+  dispatch_once(&onceToken, ^{
+    networkInfo = [[CTTelephonyNetworkInfo alloc] init];
+  });
+  return networkInfo;
+}
+#endif
+
+#pragma mark - Proto creation utilities
+
+/** Converts a dictionary of <NSString *, NSNumber *> to a GPBStringInt64Dictionary proto object.
+ *  @return Reference to a GPBStringInt64Dictionary object.
+ */
+static GPBStringInt64Dictionary *FPRGetProtoCounterForDictionary(
```

Comment:
This might not be required. For now, lets comment out its implementation.

Reply:
Updated.